app/src/main/java/mozilla/lockbox/view/SettingFragment.kt

```diff
@@ -58,6 +61,14 @@ class SettingFragment : BackableFragment(), SettingView {
         return view
     }
 
+    private fun addContentDescriptions(view: View?) {
+        view!!.settingList.getChildAt(0).title.contentDescription = R.string.auto_lock.toString()
+        view.settingList.getChildAt(1).toggle.contentDescription = R.string.autofill.toString()
+        view.settingList.getChildAt(2).toggle.contentDescription = R.string.send_usage_data.toString()
+        view.settingList.getChildAt(3).title.contentDescription = R.string.app_version.toString()
+        view.settingList.getChildAt(4).toggle.contentDescription = R.string.unlock.toString()
```

Comment:
This should likely be in the `SettingsPresenter`, in the [cell configurations][1].

[1]: https://github.com/mozilla-lockbox/lockbox-android/blob/745200963efacc617ab2479d2e50caea086bc999/app/src/main/java/mozilla/lockbox/presenter/SettingPresenter.kt#L96

Reply:
The presenter doesn't know about the content descriptions. It only has access to the SettingsView, and doesn't recognize that the titles/toggles could have content descriptions.

Reply:
@eliserichards I think James is correct here; we need to extend the `SettingCellConfiguration` to include a content description rather than hard-coding values in the `View`. The implemented solution is brittle because the individual items will change positions in the list based on whether or not the device supports fingerprint unlocking, so for the majority of devices this will result in the wrong content descriptions on toggles.

Reply:
Already talked to James. I forgot we could add it in the viewholder :)

Reply:
😬 apologies for the drive-by then!!